src/lib/PnP.Framework/Graph/SubscriptionsUtility.cs

```diff
@@ -27,24 +29,19 @@ public static Model.Subscription GetSubscription(string accessToken, Guid subscr
         {
             try
             {
-                // Use a synchronous model to invoke the asynchronous process
-                var result = Task.Run(async () =>
-                {
-                    var graphClient = GraphUtility.CreateGraphClient(accessToken, retryCount, delay, azureEnvironment: azureEnvironment);
+                var requestUrl = $"{GraphHttpClient.GetGraphEndPointUrl(azureEnvironment)}subscriptions/{subscriptionId}";
 
-                    var subscription = await graphClient.Subscriptions[subscriptionId.ToString()]
-                        .Request()
-                        .GetAsync();
+                var responseAsString = HttpHelper.MakeGetRequestForString(requestUrl, accessToken, retryCount: retryCount, delay: delay);
 
-                    var subscriptionModel = MapGraphEntityToModel(subscription);
-                    return subscriptionModel;
-                }).GetAwaiter().GetResult();
+                var response = JToken.Parse(responseAsString);
```

Comment:
@quails4Eva : would recommend using System.Text.Json over NewtonSoft

Reply:
I think I've done this. I copied the newtonsoft code from elsewhere in the project, but I think this should be equivalent for system.text.json . It needed the package version updating for .net standard. I've changed it to use the same version for each target framework as I don't think system.text.json is tied to framework versions as some other packages are